Library/AuthHandler.ts

```diff
@@ -0,0 +1,91 @@
+﻿import http = require("http");
+import https = require("https");
+import azureCore = require("@azure/core-http");
+import azureIdentity = require("@azure/identity");
+
+import Config = require("./Config");
+
+/**
+ * The automated token refresh will only start to happen at the
+ * expiration date minus the value of timeBetweenRefreshAttemptsInMs,
+ * which is by default 30 seconds.
+ */
+const timeBetweenRefreshAttemptsInMs = 30000;
+
+const applicationInsightsResource = "https://monitor.azure.com";
+
+class AuthorizationOptions {
+    appId?: string = "";
+    tenantId?: string = "";
+    certificateThumbprint?: string = "";
+    certificateSubjectName?: string = "";
+    certificateStoreLocation?: string = "";
+    appKey?: string = "";
+
+    public isCertAuth(): boolean {
+        return this.appId != "" && this.tenantId != "" && this.certificateStoreLocation != "" &&
+            (this.certificateThumbprint != "" || this.certificateSubjectName != "");
+    }
+
+    public isSecretAuth(): boolean {
+        return this.appId != "" && this.tenantId != "" && this.appKey != "";
+    }
+
+    public isManagedIdentityAuth(): boolean {
+        return this.appId != "";
+    }
+}
+
+class AuthHandler {
+
+    private _tokenCache: azureCore.AccessTokenCache;
+    private _tokenRefresher: azureCore.AccessTokenRefresher;
+
+    constructor(config: Config) {
+        let credential: azureCore.TokenCredential;
+
+        let authOptions = new AuthorizationOptions();
```

Comment:
We are using what comes from config directly. How about some basic validation?

Reply:
Not using config anymore, user need to provide TokenCredential